rcl/src/rcl/arguments.c

```diff
@@ -743,6 +750,12 @@ rcl_parse_arguments(
     }
   }
 
+  // Drop parameter overrides if none was found.
```

Comment:
Why this change in the API is needed? It's ok, but I don't understand the rationale

Reply:
To keep it consistent. Before this patch, client code would assume a parameter overrides' structure can always be queried, whether it's empty or not. And that wasn't true when `argc == 0`. That's also not what the rest of the API does e.g. check `rcl_arguments_get_param_files`.